### ISSUE TITLE
Detect runtime error when `dataclass` override removes default value

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3003]
If a dataclass defines a data field that has a default value but that dataclass is subclassed by a class that redefines the same field without a default value, it can result in a condition where a runtime exception is generated.

This false negative was recently reported against mypy, and pyright also fails to detect and report this condition. https://github.com/python/mypy/issues/12137

```python
from dataclasses import dataclass

@dataclass
class A:
    a: int = 0

@dataclass
class B(A):
    a: int
    b: str
```

**Comment by erictraut** [comment]
This will be addressed in the next version.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.219, which I just published. It will also be in the next release of pylance.
